app/controllers/api/v1/recordings_controller.rb

```diff
@@ -57,19 +57,28 @@ def resync
         render_json(status: :ok)
       end
 
-      def publish_recording
-        publish = params[:publish]
+      # POST /api/v1/recordings/update_visibility.json
+      def update_visibility
         record_id = params[:record_id]
-        case publish
-        when 'true'
-          visibility = 'Published'
-        when 'false'
-          visibility = 'Unpublished'
+        visibility = params[:visibility]
+
+        case visibility
+        when 'Published'
+          published = true
+          protected = false
+        when 'Unpublished'
+          published = false
+          protected = false
+        when 'Protected'
+          published = true
+          protected = true
         end
+
         Recording.find_by(record_id:).update(visibility:)
-        BigBlueButtonApi.new.publish_recordings(record_ids: record_id, publish:)
+        BigBlueButtonApi.new.publish_recordings(record_ids: record_id, publish: published)
```

Comment:
We can further optimise this since some expensive updates aren't always needed.
Example: From Published to Unpublished only the published attribute changes so no need to further change the protected.
Same for Published and Protected.